std-bits/base/src/main/java/org/enso/base/parser/Separators.java

```diff
@@ -0,0 +1,130 @@
+package org.enso.base.parser;
+
+import static org.enso.base.parser.NumberWithSeparators.isDigit;
+
+import java.nio.CharBuffer;
+
+public record Separators(char first, char second, int count, int endIdx, int lastSeparator) {
+  /** Check if the character is a separator. */
+  static boolean isSeparator(char c) {
+    return c == '.' || c == ',' || c == ' ' || c == '\'';
+  }
+
+  /**
+   * Strip out the specified separators and replace with just full stop for decimal. If any
+   * character other than a digit, thousands or decimal separator is encountered then return null.
+   * If multiple decimal separators are encountered then return null.
+   */
+  static CharSequence strip(CharSequence value, int idx, int endIdx, char thousands, char decimal) {
+    boolean foundDecimal = false;
+    char[] results = new char[endIdx - idx];
+    int resultIdx = 0;
+    for (int i = idx; i < endIdx; i++) {
+      char c = value.charAt(i);
+      if (c == decimal) {
+        if (foundDecimal) {
+          return null;
+        }
+        results[resultIdx++] = '.';
+        foundDecimal = true;
+      } else if (isDigit(c)) {
+        results[resultIdx++] = c;
+      } else if (c != thousands) {
+        return null;
+      }
+    }
+    return CharBuffer.wrap(results, 0, resultIdx);
+  }
+
+  private static boolean validChar(char c, char first, char second) {
+    return isDigit(c)
+        || (first == NumberWithSeparators.Constants.NONE
+            ? isSeparator(c)
+            : (second == NumberWithSeparators.Constants.NONE
+                ? c == first || c == ',' || c == '.'
+                : c == second));
+  }
+
+  /**
+   * Find the number and separators section. Validate the spacing of separators. Return the
+   * separators found or null if invalid.
+   */
+  static Separators parse(CharSequence value, int idx, boolean integer) {
+    int endIdx = idx;
+    char firstSeparator = NumberWithSeparators.Constants.NONE;
+    char secondSeparator = NumberWithSeparators.Constants.NONE;
+
+    boolean firstWasSeparator = false;
+    int lastSeparator = -1;
+    int separatorCount = 0;
+
+    // Scan the text, find and validate spacing of separators.
+    // Space and ' are both valid thousands separators, but can't be second separator.
+    // ToDo: Cope with scientific notation.
```

Comment:
nitpick but I think it would be good to keep our todos in a common format. Makes it easier to grep for them etc. Even if you plan removing it before merging this PR, just in case one of them gets forgotten.

```suggestion
    // TODO Cope with scientific notation.
```